### PR TITLE
Fix flaky test_matrix_power: seed the input and condition the negative-power case

### LINKED ISSUE #23337: Flaky test: LinalgOpsCorrectnessTest::test_matrix_power asserts atol=1e-6 on an unseeded float32 matrix inverse
`LinalgOpsCorrectnessTest::test_matrix_power` fails intermittently in CI, on unrelated PRs, across backends. The negative-power assertion compares a float32 matrix inverse against `atol=rtol=1e-6` on **unseeded** random input, so whether it passes depends on the condition number of the matrices that happen to be drawn.

## Where it has failed in CI

Four unrelated branches in the last few days. Each link goes to the exact job whose log contains `FAILED keras/src/ops/linalg_test.py::LinalgOpsCorrectnessTest::test_matrix_power`:

| date | branch | job | log |
|---|---|---|---|
| 2026-07-27 | `perf-result-type-cache` | Run tests on CPU (jax) | [job 90021543522](https://github.com/keras-team/keras/actions/runs/30279375522/job/90021543522) |
| 2026-07-27 | `iou` | Run tests on CPU (openvino) | [job 89907590427](https://github.com/keras-team/keras/actions/runs/30244201061/job/89907590427) |
| 2026-07-25 | `mlx-rebase` | Run tests on CPU (mlx) | [job 89722363555](https://github.com/keras-team/keras/actions/runs/30175050756/job/89722363555) |
| 2026-07-24 | `feat/layers/ternary-dense` | Run tests on CPU (openvino) | [job 89548909880](https://github.com/keras-team/keras/actions/runs/30113638588/job/89548909880) |

None of these branches touch `linalg`, `matrix_power`, or anything numerical.

## Minimal reproduction

Against `master` at `abd068b3cf`, no other changes:

```python
import numpy as np
from keras.src.ops import linalg

np.random.seed(31)                                # any unlucky draw will do
x = np.random.rand(4, 3, 3).astype("float32")
x_inv_stable = (x + np.eye(3)).astype("float32")  # verbatim from the test

out = np.asarray(linalg.matrix_power(x_inv_stable, -2))
expected = np.linalg.matrix_power(x_inv_stable, -2)

print("max condition number:", np.linalg.cond(x_inv_stable).max())
print("max |out - expected|:", np.abs(out - expected).max())
np.testing.assert_allclose(out, expected, atol=1e-6, rtol=1e-6)
```

| backend | max cond(x) | max abs diff | mismatched |
|---|--:|--:|--:|
| jax | 210.6 | 3.17e-03 | 4/36 |
| tensorflow | 210.6 | 3.17e-03 | 4/36 |
| torch | 210.6 | 9.52e-03 | 9/36 |

The failure is not backend-specific — it is the tolerance.

## Failure rate on unseeded input

Running the test's own negative-power branch 500 times on `master`, CPU:

| backend | runs exceeding `atol=rtol=1e-6` | worst max abs diff |
|---|--:|--:|
| jax | 4 / 500 (0.8%) | 3.47e-02 |
| tensorflow | 1 / 500 (0.2%) | 1.04e-03 |
| torch | 0 / 500 | 1.53e-04 |

Even where no run crossed the threshold, the worst observed error is two orders of magnitude above `1e-6`, so every backend is one unlucky draw away from failing.

## Traceback

From a recent CI run (jax, Python 3.11):

```
    # Negative power
    x_inv_stable = (x + np.eye(3)).astype("float32")
    out = linalg.matrix_power(x_inv_stable, -2)
    expected = np.linalg.matrix_power(x_inv_stable, -2)
>   self.assertAllClose(
        out, expected, atol=1e-6, tpu_atol=1e-2, tpu_rtol=1e-2
    )

keras/src/ops/linalg_test.py:756:
E       AssertionError:
E       Not equal to tolerance rtol=1e-06, atol=1e-06
E
E       Mismatched elements: 1 / 36 (2.78%)
E       Mismatch at index:
E        [3, 1, 0]: 1.9887009859085083 (ACTUAL), 1.9887044429779053 (DESIRED)
E       Max absolute difference among violations: 3.4570694e-06
```

## Why it happens

`matrix_power(x, -2)` inverts `x` and squares it. The forward error of a float32 inverse grows with the condition number of the input, roughly `cond(x) * eps` with `eps ≈ 1.2e-7` for float32. `np.random.rand(4, 3, 3) + np.eye(3)` routinely produces `cond` in the tens-to-hundreds — the runs above reach 271 — which puts the expected error in the `1e-5` to `1e-2` range, thousands of times looser than the `1e-6` the test asserts.

The other three assertions in the same test (powers 3, 0 and 1) involve no inversion, so they are unaffected.

## Expected behaviour

The test should pass deterministically on every run and every backend.

## Suggested fix

The same class of flakiness was already fixed for `test_lstsq` in this file, and that fix is a good template — `keras/src/ops/linalg_test.py:673`:

```python
# Seed for determinism: lstsq with `rcond` is sensitive to the
# conditioning of randomly drawn matrices, and an unseeded input
# was the source of historical flakiness on the torch backend.
rng = np.random.default_rng(0)
```

For `test_matrix_power`, seeding alone removes the randomness but still pins the assertion to whatever error that one seed happens to produce. Both parts are worth doing:

1. Seed the input, as `test_lstsq` does.
2. Use a tolerance appropriate for a float32 inverse on the negative-power assertion specifically, rather than the `1e-6` shared with the inversion-free cases. `test_svd` in the same file already sets `atol=1e-3` with the comment "High tolerance due to numerical instability", which is the right precedent.

Happy to send a PR if that approach sounds right.

### PR DESCRIPTION
**Pull request by pctablet505**
Fixes #23337

## Problem

The negative-power case in `test_matrix_power` inverts a float32 matrix and asserts `atol=rtol=1e-6` against an **unseeded** input. A float32 inverse's error scales with the condition number of the matrix, and `np.random.rand(4, 3, 3) + np.eye(3)` draws condition numbers from single digits into the hundreds — often enough to cross `1e-6` regardless of backend. Failing CI runs, root cause, and a deterministic repro are in #23337.

## Fix

- Seed the input (`np.random.default_rng(0)`), the same pattern `test_lstsq` already uses in this file.
- Raise the diagonal shift on the negative-power case from `x + np.eye(3)` to `x + 3 * np.eye(3)`, so the matrix is as well conditioned as its `x_inv_stable` name implies.

Tolerances are unchanged — conditioning the input fixes the cause instead of hiding it behind a looser assertion.

## Verification

Condition number at the seeded input: ~13.2 → ~2.1 (worst over 300 seeds: ~30.9 → ~2.3). Largest error as a fraction of the assertion tolerance, over 300 seeds:

| backend | worst diff / tolerance | margin |
|---|--:|--:|
| jax | 0.0402 | 25x |
| tensorflow | 0.0402 | 25x |
| torch | 0.0401 | 25x |
| numpy | 0.0 | exact |

`keras/src/ops/linalg_test.py` passes: jax 238, tensorflow 238, torch 238, numpy 237 (+1 skipped), openvino 205 (+33 skipped).

- [x]  I am a human, and not a bot.
- [x]  I will be responsible for responding to review comments in a timely manner.
- [x]  I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

**Review by JyotinderSingh — APPROVED** [review]
Thanks for the fixes. LGTM
